credential/manifest/model.go

```diff
@@ -212,11 +224,80 @@ func IsValidCredentialApplicationForManifest(cm CredentialManifest, ca Credentia
 			if _, err := jsonpath.Compile(submissionDescriptor.Path); err != nil {
 				return fmt.Errorf("invalid json path: %s", submissionDescriptor.Path)
 			}
+		}
+
+		// index submission descriptors by id of the input descriptor
+		submissionDescriptorLookup := make(map[string]exchange.SubmissionDescriptor)
+		for _, d := range ca.PresentationSubmission.DescriptorMap {
+			submissionDescriptorLookup[d.ID] = d
+		}
+
+		// validate each input descriptor is fulfilled
+		for _, inputDescriptor := range cm.PresentationDefinition.InputDescriptors {
+			submissionDescriptor, ok := submissionDescriptorLookup[inputDescriptor.ID]
+			if !ok {
+				return fmt.Errorf("unfulfilled input descriptor<%s>; submission not valid", inputDescriptor.ID)
+			}
+
+			// if the format on the submitted claim does not match the input descriptor, we cannot process
+			if inputDescriptor.Format != nil && !util.Contains(submissionDescriptor.Format, inputDescriptor.Format.FormatValues()) {
+				return fmt.Errorf("for input descriptor<%s>, the format of submission descriptor<%s> is not one"+
+					"  of the supported formats: %s", inputDescriptor.ID, submissionDescriptor.Format,
+					strings.Join(inputDescriptor.Format.FormatValues(), ", "))
+			}
+
+			// TODO(gabe) support nested paths in presentation submissions
+			// https://github.com/TBD54566975/ssi-sdk/issues/73
+			if submissionDescriptor.PathNested != nil {
+				return fmt.Errorf("submission with nested paths not supported: %s", submissionDescriptor.ID)
+			}
+
+			// resolve the claim from the JSON path expression in the submission descriptor
+			submittedClaim, err := jsonpath.JsonPathLookup(vcs, submissionDescriptor.Path)
+
+			if err != nil {
+				return errors.Wrapf(err, "could not resolve claim from submission descriptor<%s> with path: %s", submissionDescriptor.ID, submissionDescriptor.Path)
+			}
+
+			// convert submitted claim vc to map[string]interface{}
+			cred := submittedClaim.(credential.VerifiableCredential)
```

Comment:
does this work reliably? I always did the json marshal/unmarshal if this works it could simplify a good amount of code

Reply:
seems to work in all unit tests, I will create more tests